MyCoRe-Org/mir
# MIR-151 Integration tests with SOLR 7

# DO NOT MERGE YET.
[MIR-151 Integration tests with SOLR 7](https://mycore.atlassian.net/browse/MIR-151)

## Changes
- removed solr.war from cargo because there is no official .war
- added solr-runner plugin which:
  - downloads a specific solr
  - copies solr-config to mycore-home
  - starts and stops solr
- start/stop solr before/after integration test
- cleaned the schema.xml
  - i used a "fresh" solr configuration
  - copied all the mycore-stuff to it
  - removed most of the default stuff which we dont use in mycore

## TODO
- Maybe find a better location for solr config files (@yagee-de please comment)
- Maybe let mir-wizard copy/download the new configuration, because currently it downloads the 4.* config (@adlerre please comment)


## How to test this
- check out the right branch
- run `mvn clean install -Djetty` to run the integration tests (should work out of the Box) 
- change the url in the mycore.properties to match the new port `8983` (which is the solr standard port)  
- run `mvn solr-runner:copyHome solr-runner:start` in mir-webapp folder to copy all solr cofiguration files and start solr in the background.
